packages/eslint-plugin/src/rules/consistent-type-imports.ts

```diff
@@ -730,30 +728,25 @@ export default createRule<Options, MessageIds>({
           } else {
             afterFixes.push(insertTypeNamedSpecifiers);
           }
+        }
+        // The import is both default and named.  Insert named on new line because can't mix default type import and named type imports
+        else if (fixStyle === 'inline-type-imports') {
+          yield fixer.insertTextBefore(
+            node,
+            `import {${typeNamedSpecifiers
+              .map(spec => {
+                const insertText = context.sourceCode.text.slice(...spec.range);
+                return `type ${insertText}`;
+              })
+              .join(', ')}} from ${context.sourceCode.getText(node.source)};\n`,
+          );
         } else {
-          // The import is both default and named.  Insert named on new line because can't mix default type import and named type imports
```

Comment:
I think this is intentional, because this comment needs to apply to this whole `if...else`. I'm not sure how to help here.

Reply:
Yeah, this kind of inflexibility over expressiveness is why i don't like this rule 🙃

Reply:
ooh, good catch